snuba/datasets/events_format.py

```diff
@@ -47,20 +59,22 @@ def extract_extra_tags(
 
 
 def extract_nested(
-    nested_col: Mapping[str, Any], val_processor: Callable[[Any], TVal]
+    nested_col: Mapping[str, Any], val_processor: Callable[[Any], Optional[TVal]]
 ) -> Tuple[Sequence[str], Sequence[TVal]]:
     keys = []
     values = []
     for key, value in sorted(nested_col.items()):
         value = val_processor(value)
         if value is not None:
-            keys.append(_unicodify(key))
+            keys.append(cast(str, _unicodify(key)))
```

Comment:
Should we not deal with the case where unicodify returns None?
Like asserting before that we only accept valid strings ?

Reply:
`unicodify` can return None but only if None is passed, which it is not in this case. Not sure if there's a better way to handle this in unicodify.